BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -254,6 +264,39 @@ public class AppProcessor {
             services.errorReporter.log(error: error)
         }
     }
+
+    /// Starts timer to send organization events regularly
+    private func startEventTimer() {
+        sendEventTimer = Timer.scheduledTimer(withTimeInterval: 5 * 60, repeats: true) { _ in
+            Task { [weak self] in
+                await self?.services.eventService.upload()
+            }
+        }
+        sendEventTimer?.tolerance = 10
+    }
+
+    /// Stops the timer for organization events
+    private func stopEventTimer() {
+        if let taskId = backgroundTaskId {
+            services.application?.endBackgroundTask(taskId)
+            backgroundTaskId = nil
+        }
+        backgroundTaskId = services.application?.beginBackgroundTask(
+            withName: "SendEventBackgroundTask",
+            expirationHandler: { [weak self] in
+                if let backgroundTaskId = self?.backgroundTaskId {
+                    self?.services.application?.endBackgroundTask(backgroundTaskId)
+                    self?.backgroundTaskId = nil
+                }
+            }
+        )
+        sendEventTimer?.fire()
+        sendEventTimer?.invalidate()
+        if let taskId = backgroundTaskId {
+            services.application?.endBackgroundTask(taskId)
```

Comment:
🤔 Since the upload API call is made asynchronously, is it possible/likely that the background task is ended before the upload completes? Would it be better to move this to after the request completes?

Reply:
Ah, good point.